packages/@tailwindcss-upgrade/src/migrate.ts

```diff
@@ -122,3 +128,211 @@ export async function analyze(stylesheets: Stylesheet[]) {
     }
   }
 }
+
+export async function split(stylesheets: Stylesheet[]) {
+  let stylesheetsById = new Map<StylesheetId, Stylesheet>()
+  for (let sheet of stylesheets) {
+    stylesheetsById.set(sheet.id, sheet)
+  }
+
+  // Keep track of sheets that contain `@utillity` rules
+  let containsUtilities = new Set<Stylesheet>()
+
+  for (let sheet of stylesheets) {
+    let isLayered = sheet.layers.has('utilities') || sheet.layers.has('components')
+    if (!isLayered) continue
+
+    walk(sheet.root, (node) => {
+      if (node.type !== 'atrule') return
+      if (node.name !== 'utility') return
+
+      containsUtilities.add(sheet)
+
+      return WalkAction.Stop
+    })
+  }
+
+  // Split every imported stylesheet into two parts
+  let utilitySheets = new Map<Stylesheet, Stylesheet>()
+
+  for (let sheet of stylesheets) {
+    // Ignore stylesheets that were not imported
+    if (!sheet.file) continue
+    if (sheet.parents.size === 0) continue
+
+    // Skip stylesheets that don't have utilities
+    // and don't have any children that have utilities
+    if (!containsUtilities.has(sheet)) {
+      if (!Array.from(sheet.descendants).some((child) => containsUtilities.has(child))) {
+        continue
+      }
+    }
+
+    let utilities = postcss.root({
+      raws: {
+        tailwind_pretty: true,
+      },
+    })
+
+    walk(sheet.root, (node) => {
+      if (node.type !== 'atrule') return
+      if (node.name !== 'utility') return
+
+      // `append` will move this node from the original sheet
+      // to the new utilities sheet
+      utilities.append(node)
+
+      return WalkAction.Skip
+    })
+
+    let utilitySheet = await Stylesheet.fromRoot(
+      utilities,
+      sheet.file.replace(/\.css$/, '.utilities.css'),
```

Comment:
A file with this name can already exist. If it does, we should error with something useful

Reply:
This has been addressed